classes/user/maps/Schema.inc.php

```diff
@@ -0,0 +1,262 @@
+ */
+
+namespace PKP\user\maps;
+
+use APP\core\Request;
+use APP\facades\Repo;
+use Illuminate\Support\Enumerable;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\plugins\HookRegistry;
+use PKP\services\PKPSchemaService;
+use PKP\user\User;
+use Psr\Http\Message\RequestInterface as SlimRequest;
+
+class Schema extends \PKP\core\maps\Schema
+{
+    /** @var Enumerable */
+    public $collection;
+
+    /** @var string */
+    public $schema = PKPSchemaService::SCHEMA_USER;
+
+    public function __construct(Request $request, Context $context, PKPSchemaService $schemaService)
+    {
+        parent::__construct($request, $context, $schemaService);
+    }
+
+    /**
+     * Map a publication
+     *
+     * Includes all properties in the user schema.
+     */
+    public function map(User $item, ?SlimRequest $slimRequest = null): array
+    {
+        return $this->mapByProperties($this->getProps(), $item, $slimRequest);
+    }
+
+    /**
+     * Summarize a user
+     *
+     * Includes properties with the apiSummary flag in the user schema.
+     */
+    public function summarize(User $item, ?SlimRequest $slimRequest = null): array
+    {
+        return $this->mapByProperties($this->getSummaryProps(), $item, $slimRequest);
+    }
+
+    /**
+     * Summarize a user with reviewer data
+     *
+     * Includes properties with the apiSummary flag in the user schema.
+     */
+    public function summarizeReviewer(User $item, ?SlimRequest $slimRequest = null): array
+    {
+        return $this->mapByProperties(array_merge($this->getSummaryProps(), ['reviewsActive', 'reviewsCompleted', 'reviewsDeclined', 'reviewsCancelled', 'averageReviewCompletionDays', 'dateLastReviewAssignment', 'reviewerRating']), $item, $slimRequest);
+    }
+
+    /**
+     * Map a collection of Users
+     *
+     * @see self::map
+     */
+    public function mapMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->map($item);
+        });
+    }
+
+    /**
+     * Summarize a collection of users
+     *
+     * @see self::summarize
+     */
+    public function summarizeMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->summarize($item);
+        });
+    }
+
+    /**
+     * Map schema properties of a user to an assoc array
+     */
+    protected function mapByProperties(array $props, User $user, ?SlimRequest $slimRequest = null): array
+    {
+        $output = [];
+
+        foreach ($props as $prop) {
+            switch ($prop) {
+                case 'id':
+                    $output[$prop] = (int) $user->getId();
+                    break;
+                case 'userName':
+                    $output[$prop] = $user->getUserName();
+                    break;
```

Comment:
I've been avoiding mapping every property manually by having a default condition call getData:

```php
   default:
        $output[$prop] = $user->getData($prop);
        break;
```

That way we only have to write code when mapping something that isn't straightforward.

Reply:
Done: https://github.com/asmecher/pkp-lib/commit/a9434fa2cc7a48e7db99e50ecf963b203388010a